wsmaster/che-core-api-installer/src/main/java/org/eclipse/che/api/installer/server/jpa/JpaInstallerDao.java

```diff
@@ -0,0 +1,176 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2017 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.installer.server.jpa;
+
+import com.google.inject.Singleton;
+import com.google.inject.persist.Transactional;
+
+import org.eclipse.che.api.core.Page;
+import org.eclipse.che.api.installer.server.exception.InstallerConflictException;
+import org.eclipse.che.api.installer.server.exception.InstallerException;
+import org.eclipse.che.api.installer.server.exception.InstallerNotFoundException;
+import org.eclipse.che.api.installer.server.impl.InstallerFqn;
+import org.eclipse.che.api.installer.server.model.impl.InstallerImpl;
+import org.eclipse.che.api.installer.server.spi.InstallerDao;
+import org.eclipse.che.api.installer.shared.model.Installer;
+import org.eclipse.che.core.db.DBInitializer;
+import org.eclipse.che.core.db.jpa.DuplicateKeyException;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.persistence.EntityManager;
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * @author Anatolii Bazko
+ */
+@Singleton
+public class JpaInstallerDao implements InstallerDao {
+
+    private final Provider<EntityManager> managerProvider;
+
+    @Inject
+    public JpaInstallerDao(@SuppressWarnings("unused") DBInitializer dbInitializer,
```

Comment:
Have you investigated why other Daos like JpaStackDao, JpaWorkspace, JpaAccountDao, ... ?